aspnetcore/grpc/versioning.md

```diff
@@ -0,0 +1,98 @@
+
+Making non-breaking changes to a service has a number of benefits:
+
+- Existing clients continue to run.
+- Avoids work involved with notifying clients of breaking changes, and updating them.
+- Only one version of the service needs to be documented and maintained.
+
+This content focuses on whether changes are **breaking at a gRPC protocol and .NET binary compatibility level**. When making changes, consider whether older clients can logically continue working. For example, adding a new field to a request message:
+
+* Is not a protocol breaking change.
+* Returning an error status on the server if the new field is not set makes it a breaking change for old clients.
+
+### Non-breaking changes
+
+These changes are non-breaking at a gRPC protocol level, and .NET binary level.
+
+- **Adding a new service**
+- **Adding a new method to a service**
+- **Adding a field to a request message** - Fields added to a request message are deserialized with the [default value](https://developers.google.com/protocol-buffers/docs/proto3#default) on the server when not set. To be non-breaking the service will need to succeed when it is not set by older clients.
+- **Adding a field to a response message** - Fields added to a response message are deserialized into the message's [unknown fields](https://developers.google.com/protocol-buffers/docs/proto3#unknowns) collection on the client.
+- **Adding a value to an enum** - Enums are serialized as a numeric value. New enum values are deserialized on the client to the enum value without an enum name. To be non-breaking older clients will need to run correctly when they receive and unexcepted value.
+
+### Binary breaking changes
+
+The following changes are non-breaking at a gRPC protocol level, but the client needs to be updated if it upgrades to the latest *.proto* contract or client .NET assembly. Binary compatibility is important if you plan to publish a gRPC library to NuGet.
+
+- **Removing a field** - Values from a removed field are deserialized to a message's [unknown fields](https://developers.google.com/protocol-buffers/docs/proto3#unknowns). This isn't a gRPC protocol breaking change, but the client needs to be updated if it upgrades to the latest contract. It is important that a removed field number isn't accidentally reused in the future. One way to make sure this doesn't happen is to specify deleted field numbers and names on the message using Protobuf's [`reserved`](https://developers.google.com/protocol-buffers/docs/proto3#reserved) keyword.
+- **Renaming a field** - Field names are only used in generated code. The field number is used to identify fields on the network. The client will need to be updated if it upgrades to the latest contract.
```

Comment:
renaming a field is discouraged. If JSON formatting or reflection is used, things can potentially break - basically, it's a thin ice. 
same for renaming messages (this can also affect the Any type).